.github/workflows/component-template-e2e-tests.yml

```diff
@@ -0,0 +1,96 @@
+name: Component-template repository
+
+on:
+  push:
+    branches:
+      - "develop"
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+jobs:
+  e2e-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Streamlit code
+        uses: actions/checkout@v3
+        with:
+          persist-credentials: false
+          submodules: "recursive"
+          fetch-depth: 2
+
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: "19.x"
+          cache: "yarn"
+          cache-dependency-path: "component-lib/yarn.lock"
+
+      - name: Set Python version vars
+        uses: ./.github/actions/build_info
+
+      - name: Set up Python ${{ env.PYTHON_MAX_VERSION }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: "${{ env.PYTHON_MAX_VERSION }}"
+
+      - name: Setup virtual env
+        uses: ./.github/actions/make_init
+
+      - name: Run make develop
+        run: make develop
+
+      - name: Build Package
+        timeout-minutes: 120
+        run: |
+          sudo apt install rsync
+          make package
```

Comment:
nit: Unfortunately, this step takes an immense amount of time. I think we can significantly get the time down with the same result by running these commands instead of `make package`:

```
make frontend-fast
make distribution
```

Reply:
Great, thanks!

Reply:
Btw. another alternative might be:
```
BUILD_AS_FAST_AS_POSSIBLE=1 make package
```